src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -154,6 +186,67 @@ public IObservable<Unit> Push(ILocalRepositoryModel repository)
             });
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2201", Justification = "Prototype")]
+        public async Task<bool> SyncSubmodules(ILocalRepositoryModel repository, IProgress<string> progress)
+        {
+            var exitCode = Where("git");
+            if (exitCode != 0)
+            {
+                progress.Report(App.Resources.CouldntFindGitOnPath);
+                return false;
+            }
+
+            return await SyncSubmodules(repository.LocalPath, progress) == 0;
+        }
+
+        // LibGit2Sharp has limited submodule support so shelling out Git.exe for submodule commands.
+        async Task<int> SyncSubmodules(string workingDir, IProgress<string> progress)
+        {
+            var cmdArguments = "/C git submodule init & git submodule sync --recursive & git submodule update --recursive";
+            var startInfo = new ProcessStartInfo("cmd", cmdArguments)
+            {
+                WorkingDirectory = workingDir,
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+
+            using (var process = Process.Start(startInfo))
+            {
+                await Task.WhenAll(
+                    ReadLinesAsync(process.StandardOutput, progress),
+                    ReadLinesAsync(process.StandardError, progress),
+                    Task.Run(() => process.WaitForExit()));
+                return process.ExitCode;
+            }
+        }
+
+        static int Where(string fileName)
+        {
+            var cmdArguments = "/C WHERE /Q " + fileName;
+            var startInfo = new ProcessStartInfo("cmd", cmdArguments)
+            {
+                UseShellExecute = false,
+                CreateNoWindow = true
+            };
+
+            using (var process = Process.Start(startInfo))
+            {
+                process.WaitForExit();
```

Comment:
This will block the UI thread waiting for the process to exit, probably not a good idea...